### PR TITLE
Donor can register account

### PR DESCRIPTION
**Pull request by swardbladpaulo**
[DONOR: Donor can register for an account](https://www.pivotaltracker.com/story/show/176623667)
### User Story
```
As an Donor,
In order to access the application,
I would like the site to have a unique way for a donor to register
```
## Changes proposed in this pull request:
* create test for donor can register account
* add fixture, adjust registration form to be trigger, create registration form, add new state to rootreducer
* registration button on right place, add background image
* Authentication continues with reg form
* Adding a error message for registration
* Adding j-tockauth
* Adding error message
* add axios, switch order on submit button, import axios to index.js, modify registrationform: add currentuser message, removed onclick event, add name to inputs, modify auth set current user added data to payload

## What have we learned working on this feature:
* working more with debuggers
* axios
* authentication
*  the importance of the order on the test
* name conventions on test
* 
## Packages/Gems added
* J-tockAuth
* Axios

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,11 @@
     "@testing-library/jest-dom": "^5.11.4",
     "@testing-library/react": "^11.1.0",
     "@testing-library/user-event": "^12.1.10",
+    "axios": "^0.21.1",
+    "j-tockauth": "^1.2.7",
+    "npm": "^6.14.11",
     "react": "^17.0.1",
+    "react-axios": "^2.0.4",
```
Comment (FlisAnn) [inline package.json]: Do we need two axios?

Comment (Ehugo2000) [inline package.json]: wont need that i guess

File: src/App.jsx [hunk 2]
```diff
@@ -1,17 +1,16 @@
 import React from 'react'
-import { useSelector } from 'react-redux'
 import MenuHeader from './components/MenuHeader'
 import Footer from './components/Footer'
 import './index.css'
 import MainPageDisplay from './components/MainPageDisplay'
 
 const App = () => {
-  const message = useSelector((state) => state.greeting)
+
```
Comment (FlisAnn) [inline src/App.jsx]: Formatting

**Review by FlisAnn — CHANGES_REQUESTED** [review]
Looks great

**Review by Ehugo2000 — COMMENTED** [review]
only one comment

File: package.json (hunk 1 above)
Comment (swardbladpaulo) [reply package.json]: No, removed react-axios

File: src/App.jsx (hunk 2 above)
Comment (swardbladpaulo) [reply src/App.jsx]: Fixed

File: package.json (hunk 1 above)
Comment (swardbladpaulo) [reply package.json]: Removed

**Review by FlisAnn — APPROVED** [review]

**Pull request merged**